src/libpsc/psc_output_fields/fields_item_fields.hxx

```diff
@@ -204,4 +257,36 @@ private:
   MfieldsState& mflds_;
 };
 
+// ======================================================================
+// Item_grad
+
+template <typename Mfields>
+class Item_grad : public MFexpression<Item_grad<Mfields>>
+{
+public:
+  using Real = typename Mfields::real_t;
+
+  static char const* name() { return "grad"; }
+  static int n_comps() { return 3; }
+  static std::vector<std::string> comp_names()
+  {
+    return {"gradx", "grady", "gradz"};
+  }
+
+  Item_grad(Mfields& mflds) : mflds_{mflds} {}
+
+  const Grid_t& grid() const { return mflds_.grid(); }
+  Int3 ibn() const { return {}; }
+  int n_patches() const { return grid().n_patches(); }
+
+  auto gt() const
+  {
+    return psc::item::grad_ec(mflds_.gt().view(_all, _all, _all, _all),
```

Comment:
I think the `.view(_all, _all, _all, _all)` doesn't really do anything, does it?